Homework9/index.js

```diff
@@ -0,0 +1,146 @@
+const express = require("express");
+const nunjucks = require("nunjucks");
+const mongoose = require("mongoose");
+const session = require("express-session");
+const path = require("path");
+const routers = require("./routers");
+const fs = require("fs");
+const MongoStore = require("connect-mongo")(session);
+const passport = require("passport");
+const { Strategy: LocalStrategy } = require("passport-local");
+const User = require("./routers/api/models/user.model");
+const { compareSync } = require("bcryptjs");
+require("dotenv").config();
+
+const server = express();
+
+nunjucks.configure(path.join(__dirname, "templates"), {
+    watch: true,
+    autoescape: true,
+    express: server
+});
+
+mongoose.connect("mongodb://localhost:27017/telephone-book-1", {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    useCreateIndex: true
+});
+
+mongoose.connection.on("error", err => {
+    console.error("Mongo error:", err);
+    process.exit(1);
+});
+
+mongoose.set("debug", true);
+
+server.use(
+    session({
+        name: "session",
+        secret: "SecretKey1234567890",
+        resave: false,
+        saveUninitialized: false,
+        maxAge: 24 * 60 * 60 * 1000,
+        store: new MongoStore({
+            collection: "session",
+            mongooseConnection: mongoose.connection,
+            stringify: false
+        })
+    })
+);
+
+server.use(express.json());
+server.use(express.urlencoded({ extended: true }));
+server.use(express.static(path.join(__dirname, "static")));
+server.use(passport.initialize());
+server.use(passport.session());
+
+passport.use(
+    "local",
+    new LocalStrategy(
+        {
+            passReqToCallback: true,
+            passwordField: "password",
+            usernameField: "email"
+        },
+        async (req, email, password, callbackfn) => {
+            const user = await User.findOne({ email: email });
+
+            if (!user) {
+                return callbackfn(null, null);
+            }
+
+            if (!compareSync(password, user.password)) {
+                return callbackfn(null, null);
+            }
+
+            return callbackfn(null, user);
+        }
+    )
+);
+
+passport.serializeUser((req, user, callbackfn) => {
+    return callbackfn(null, user._id);
+});
+
+passport.deserializeUser(async (req, id, callbackfn) => {
+    const user = await User.findById(id);
```

Comment:
```suggestion
    const user = await User.findById(id).lean().exec();
```
А вообще, может не весь объект пользователя тебе надо в памяти?